frontend/packages/db-structure/src/parser/schemarb/index.test.ts

```diff
@@ -220,5 +220,28 @@ describe(processor, () => {
 
       expect(result.relationships).toEqual(expected)
     })
+
+    it('unique foreign key', async () => {
+      const result = await processor(/* Ruby */ `
+        create_table "posts" do |t|
+          t.bigint "user_id", unique: true
```

Comment:
I believe this test case is necessary:

```suggestion
          t.bigint "user_id"
          t.index ["user_id"], name: "index_posts_on_user_id", unique: true
```

This is a real-world example of a `has_one` association. Rails' schema dumper will be expected to place `unique: true` in the `t.index` declaration.



Rails schema dumper output `db/schema.rb`:

https://github.com/mastodon/mastodon/blob/fbcd14ff7e18c8689426433aa5ad673c9ebd6409/db/schema.rb#L272  
https://github.com/mastodon/mastodon/blob/fbcd14ff7e18c8689426433aa5ad673c9ebd6409/db/schema.rb#L281


Model (for reference):

https://github.com/mastodon/mastodon/blob/fbcd14ff7e18c8689426433aa5ad673c9ebd6409/app/models/account_warning.rb#L38

Reply:
@hoshinotsuyoshi  Thank you for information!

I thought Rails schema dumper don't care `has_one` or `has_many`.
Is this a standard Rails specification?

Reply:
@sasamuku 

> Rails schema dumper don't care has_one or has_many.

You're absolutely right. Rails schema dumper doesn't care about `has_one` or `has_many`. My description in the model was just a reference. 🙏

Reply:
@hoshinotsuyoshi 
I see! Thanks.

`t.index` can not support this implementation for now.
And we may need to care about postgres parser as well.

So let me make another pr for supporting index.

Reply:
@sasamuku thanks!